nncf/tensor/functions/tf_numeric.py

```diff
@@ -0,0 +1,619 @@
+    if mask is None:
+        return numeric.median(x, axis=axis, keepdims=keepdims)
+
+    if isinstance(axis, list):
+        axis = tuple(axis)
+
+    masked_x = tf.where(mask, np.nan, x)
+    np_masked_x = masked_x.numpy()
+    np_masked_median = np.nanquantile(np_masked_x, 0.5, axis=axis, keepdims=keepdims)
+
+    with tf.device(x.device):
+        ret = tf.constant(np_masked_median)
+        ret = tf.where(tf.math.is_nan(ret), tf.zeros_like(ret), ret)
+
+        return ret
+
+
+@numeric.expand_dims.register
+def expand_dims(a: tf.Tensor, axis: Union[int, Tuple[int, ...]]) -> tf.Tensor:
+    axes_tuple: Tuple[int, ...]
+    if isinstance(axis, int):
+        axes_tuple = (axis,)
+    elif isinstance(axis, tuple):
+        axes_tuple = axis
+    else:
+        error_msg = f"axis must be int or tuple, got {type(axis)}"
+        raise TypeError(error_msg)
+
+    if len(set(axes_tuple)) != len(axes_tuple):
+        error_msg = "repeated axis"
+        raise ValueError(error_msg)
+
+    out_ndim = len(axes_tuple) + a.ndim
+
+    norm_axis = []
+    for ax in axes_tuple:
+        if ax < -out_ndim or ax >= out_ndim:
+            error_msg = f"axis {ax} is out of bounds for array of dimension {out_ndim}"
+            raise ValueError(error_msg)
+        norm_axis.append(ax + out_ndim if ax < 0 else ax)
+
+    shape_it = iter(a.shape)
+    shape = [1 if ax in norm_axis else next(shape_it) for ax in range(out_ndim)]
+    return tf.reshape(a, shape)
+
+
+@numeric.clone.register
+def _(a: tf.Tensor) -> tf.Tensor:
+    with tf.device(a.device):
+        return tf.identity(a)
+
+
+@numeric.searchsorted.register
+def searchsorted(
+    a: tf.Tensor, v: tf.Tensor, side: Literal["left", "right"] = "left", sorter: Optional[tf.Tensor] = None
+) -> tf.Tensor:
+    if side not in ["right", "left"]:
+        error_msg = f"Invalid value for 'side': {side}. Expected 'right' or 'left'."
+        raise ValueError(error_msg)
+    if a.ndim != 1:
+        error_msg = f"Input tensor 'a' must be 1-D. Received {a.ndim}-D tensor."
+        raise ValueError(error_msg)
+    sorted_a = tf.sort(a)
+    return tf.searchsorted(sorted_sequence=sorted_a, values=v, side=side)
+
+
+def zeros(
+    shape: Tuple[int, ...],
+    *,
+    dtype: Optional[TensorDataType] = None,
+    device: Optional[TensorDeviceType] = None,
+) -> tf.Tensor:
+    tf_dtype = DTYPE_MAP[dtype] if dtype is not None else None
+    tf_device = DEVICE_MAP[device] if device is not None else None
+    with tf.device(tf_device):
+        return tf.zeros(shape, dtype=tf_dtype)
+
+
+def eye(
+    n: int,
+    m: Optional[int] = None,
+    *,
+    dtype: Optional[TensorDataType] = None,
+    device: Optional[TensorDeviceType] = None,
+) -> tf.Tensor:
+    tf_dtype = DTYPE_MAP[dtype] if dtype is not None else None
+    tf_device = DEVICE_MAP[device] if device is not None else None
+    p_args = (n,) if m is None else (n, m)
+    with tf.device(tf_device):
+        return tf.eye(*p_args, dtype=tf_dtype)
```

Comment:
This should resolve `test_fn_eye()` device problem
```suggestion
        eye = tf.eye(*p_args, dtype=tf_dtype)
        return tf.identity(eye)
```